### PR TITLE
log refactor: dm-worker

### PR DESCRIPTION
**Pull request by GregoryIan**
### What problem does this PR solve? 
refactor log of dm-worker

### What is changed and how it works?
use pingcap/log

### CODE REVIEW

File: dm/worker/server.go [hunk 1]
```diff
@@ -369,22 +371,22 @@ func (s *Server) OperateRelay(ctx context.Context, req *pb.OperateRelayRequest)
...
 
 // UpdateRelayConfig updates config for relay and (dm-worker)
 func (s *Server) UpdateRelayConfig(ctx context.Context, req *pb.UpdateRelayRequest) (*pb.CommonWorkerResponse, error) {
-	log.Infof("[server] receive UpdateRelayConfig request %+v", req)
+	log.L().Info("", zap.String("request", "update relay config"), zap.Stringer("payload", req))
 
 	err := s.worker.UpdateRelayConfig(ctx, req.Content)
 	if err != nil {
-		log.Errorf("[server] %v UpdateRelayConfig error %v", req, errors.ErrorStack(err))
+		log.L().Error("fail to Update relay configv", zap.String("request", "update relay config"), zap.Stringer("payload", req), zap.Error(err))
```
Comment (amyangfei) [inline dm/worker/server.go]:
```suggestion
		log.L().Error("fail to Update relay config", zap.String("request", "update relay config"), zap.Stringer("payload", req), zap.Error(err))
```

File: dm/worker/server.go [hunk 2]
```diff
@@ -402,15 +405,15 @@ func (s *Server) QueryWorkerConfig(ctx context.Context, req *pb.QueryWorkerConfi
...
+		log.L().Error("fail to query worker config", zap.String("request", "query worker config"), zap.Stringer("payload", req), zap.Error(err))
 		return resp, nil
 	}
 
 	rawConfig, err := workerCfg.From.Toml()
 	if err != nil {
 		resp.Result = false
 		resp.Msg = errors.ErrorStack(err)
-		log.Errorf("[worker] marshal worker config error %v", errors.ErrorStack(err))
+		log.L().Error("fail to marshal worker config", zap.String("request", "query worker config"), zap.Stringer("wotker from config", &workerCfg.From), zap.Error(err))
```
Comment (amyangfei) [inline dm/worker/server.go]:
```suggestion
		log.L().Error("fail to marshal worker config", zap.String("request", "query worker config"), zap.Stringer("worker from config", &workerCfg.From), zap.Error(err))
```
`zap.Stringer("worker from config", &workerCfg.From)` database password will be logged here ?

File: dm/worker/server.go [hunk 3]
```diff
@@ -420,11 +423,11 @@ func (s *Server) QueryWorkerConfig(ctx context.Context, req *pb.QueryWorkerConfi
 
 // MigrateRelay migrate relay to original binlog pos
 func (s *Server) MigrateRelay(ctx context.Context, req *pb.MigrateRelayRequest) (*pb.CommonWorkerResponse, error) {
-	log.Infof("[server] receive MigrateRelay request %+v", req)
+	log.L().Info("", zap.String("request", "migrate relay"), zap.Stringer("payload", req))
 
 	err := s.worker.MigrateRelay(ctx, req.BinlogName, req.BinlogPos)
 	if err != nil {
-		log.Errorf("[worker] %v MigrateRelay error %v", req, errors.ErrorStack(err))
+		log.L().Error("fail to  migrate relay", zap.String("request", "migrate relay"), zap.Stringer("payload", req), zap.Error(err))
```
Comment (amyangfei) [inline dm/worker/server.go]:
```suggestion
		log.L().Error("fail to migrate relay", zap.String("request", "migrate relay"), zap.Stringer("payload", req), zap.Error(err))
```

File: dm/worker/subtask.go [hunk 4]
```diff
@@ -245,18 +249,17 @@ retry:
...
 				st.setCurrUnit(nu)
 				// NOTE: maybe need a Lock mechanism for sharding scenario
 				st.run() // re-run for next process unit
 			}
 		case pb.Stage_Stopped:
 		case pb.Stage_Paused:
 			for _, err := range result.Errors {
-				log.Errorf("[subtask] %s dm-unit %s process error with type %v:\n %v",
-					st.cfg.Name, cu.Type(), err.Type, err.Msg)
+				st.l.Error("unit process error", zap.Stringer("unit", cu.Type()), zap.Reflect("error", err))
```
Comment (amyangfei) [inline dm/worker/subtask.go]:
```suggestion
				st.l.Error("unit process error", zap.Stringer("unit", cu.Type()), zap.Error(err))
```
`zap.Error(err)` is more simple ?

File: dm/worker/worker.go [hunk 5]
```diff
@@ -772,15 +774,15 @@ Loop:
...
+				w.l.Info("updateed sub task", zap.String("task", opLog.Task.Name), zap.Stringer("new config", taskCfg))
 				err = st.Update(taskCfg)
 			case pb.TaskOp_Stop:
 				if !exist {
 					err = errors.NotFoundf("sub task with name %s", opLog.Task.Name)
 					break
 				}
 
-				log.Infof("[worker] stop sub task %s", opLog.Task.Name)
+				w.l.Info(" stopped sub task", zap.String("task", opLog.Task.Name))
```
Comment (amyangfei) [inline dm/worker/worker.go]:
```suggestion
				w.l.Info("stop sub task", zap.String("task", opLog.Task.Name))
```

File: dm/worker/worker.go [hunk 6]
```diff
@@ -789,19 +791,19 @@ Loop:
 					break
 				}
 
-				log.Infof("[worker] pause sub task %s", opLog.Task.Name)
+				w.l.Info("paused sub task", zap.String("task", opLog.Task.Name))
```
Comment (amyangfei) [inline dm/worker/worker.go]:
```suggestion
				w.l.Info("pause sub task", zap.String("task", opLog.Task.Name))
```

File: dm/worker/server.go (hunk 2 above)
Comment (GregoryIan) [reply dm/worker/server.go]: you can check it, Stringer would call json marshal

File: dm/worker/subtask.go (hunk 4 above)
Comment (GregoryIan) [reply dm/worker/subtask.go]: it‘s not real error

File: dm/worker/worker.go (hunk 6 above)
Comment (GregoryIan) [reply dm/worker/worker.go]: already done?

File: cmd/dm-worker/main.go [hunk 7]
```diff
@@ -24,6 +25,7 @@ import (
 	"github.com/pingcap/dm/pkg/log"
 	"github.com/pingcap/dm/pkg/utils"
 	"github.com/pingcap/errors"
+	"go.uber.org/zap"
```
[the remark this answers is not in the document]
Comment (GregoryIan) [reply cmd/dm-worker/main.go]: we would refine import path in other PR for all files

File: dm/master/server.go [hunk 8]
```diff
@@ -159,7 +161,7 @@ func (s *Server) Close() {
 	}
 	err := s.rootLis.Close()
 	if err != nil && !common.IsErrNetClosing(err) {
-		log.Errorf("[server] close net listener with error %s", err.Error())
+		log.L().Error("close net listener", zap.Error(err))
```
[the remark this answers is not in the document]
Comment (GregoryIan) [reply dm/master/server.go]: no, use `zap.Error` exclusively here

File: dm/worker/subtask.go [hunk 9]
```diff
@@ -384,9 +387,9 @@ func (st *SubTask) Result() *pb.ProcessResult {
 
 // Close stops the sub task
 func (st *SubTask) Close() {
-	log.Infof("[subtask] %s is closing", st.cfg.Name)
+	st.l.Info("closing")
```
Comment (WangXiangUSTC) [inline dm/worker/subtask.go]: add task name?

File: dm/worker/subtask.go [hunk 10]
```diff
@@ -384,9 +387,9 @@ func (st *SubTask) Result() *pb.ProcessResult {
...
 	if st.cancel == nil {
-		log.Infof("[subtask] not run yet, no need to close")
+		st.l.Info("not run yet, no need to close")
```
Comment (WangXiangUSTC) [inline dm/worker/subtask.go]: ditto

File: dm/worker/subtask.go [hunk 11]
```diff
@@ -628,7 +631,7 @@ func (st *SubTask) unitTransWaitCondition() error {
 	pu := st.PrevUnit()
 	cu := st.CurrUnit()
 	if pu != nil && pu.Type() == pb.UnitType_Load && cu.Type() == pb.UnitType_Sync {
-		log.Infof("[subtask] %s wait condition between %s and %s", st.cfg.Name, pu.Type(), cu.Type())
+		st.l.Info("wait condition between two units", zap.Stringer("previous unit", pu.Type()), zap.Stringer("unit", cu.Type()))
```
Comment (WangXiangUSTC) [inline dm/worker/subtask.go]: ditto

File: dm/worker/worker.go [hunk 12]
```diff
@@ -69,6 +71,7 @@ func NewWorker(cfg *Config) (*Worker, error) {
 		relayHolder: NewRelayHolder(cfg),
 		tracer:      tracing.InitTracerHub(cfg.Tracer),
 		subTasks:    make(map[string]*SubTask),
+		l:           log.With(zap.String("component", "worker controller")),
```
Comment (WangXiangUSTC) [inline dm/worker/worker.go]: seems have `component`, `unit`, `feature` three levels, can you describe the difference between them？

File: dm/worker/worker.go [hunk 13]
```diff
@@ -586,7 +589,7 @@ func (w *Worker) UpdateRelayConfig(ctx context.Context, content string) error {
 		}
 	}
 
-	log.Info("[worker] update relay configure in subtasks success.")
+	w.l.Info("[worker] update relay configure in subtasks success.")
```
Comment (WangXiangUSTC) [inline dm/worker/worker.go]:
```suggestion
	w.l.Info("update relay configure in subtasks success")
```

File: dm/worker/worker.go [hunk 14]
```diff
@@ -772,15 +774,15 @@ Loop:
 					break
 				}
 
-				log.Infof("[worker] update sub task %s with config: %v", opLog.Task.Name, taskCfg)
+				w.l.Info("updateed sub task", zap.String("task", opLog.Task.Name), zap.Stringer("new config", taskCfg))
```
Comment (WangXiangUSTC) [inline dm/worker/worker.go]:
```suggestion
				w.l.Info("updated sub task", zap.String("task", opLog.Task.Name), zap.Stringer("new config", taskCfg))
```

File: cmd/dm-worker/main.go [hunk 15]
```diff
@@ -58,15 +65,24 @@ func main() {
...
 
 	err = s.Start()
 	if err != nil {
-		log.Errorf("start dm-worker err %s", err)
+		log.L().Error("fail to start dm-worker", zap.Error(err))
 		os.Exit(2)
 	}
 	s.Close() // wait until closed
-	log.Info("dm-worker exit")
+	log.L().Info("dm-master exit")
```
Comment (amyangfei) [inline cmd/dm-worker/main.go]:
```suggestion
	log.L().Info("dm-worker exit")
```

**Comment by GregoryIan** [comment]
@amyangfei I found this log, but not found it, real old driver

**Pull request merged**